community/kernel/src/main/java/org/neo4j/kernel/api/proc/ProcedureSignature.java

```diff
@@ -174,14 +177,26 @@ public int hashCode()
     }
 
     public ProcedureSignature( ProcedureName name,
-                               List<FieldSignature> inputSignature,
-                               List<FieldSignature> outputSignature,
-                               Mode mode )
+            List<FieldSignature> inputSignature,
+            List<FieldSignature> outputSignature,
+            Mode mode,
+            boolean deprecated,
```

Comment:
Couldn't this take in some object instead, something like

```
class Deprecation
{
   String depecatedBy(){...}
}
```
and then `ProcedureSignature` could take an `Optional<Deprecation>`